tests/unit-tests/gconstruct/test_transform.py

```diff
@@ -641,6 +644,62 @@ def test_custom_label_processor():
     assert_equal(ret[stats_info_key][1], vals)
     assert_equal(ret[stats_info_key][2], counts)
 
+
+def test_custom_edge_label_processor():
+    # test generating labels on link prediction
+    train_idx = tuple((i, j) for i in range(1, 10) for j in range(1, 10))
+    val_idx = tuple((i, j) for i in range(11, 14) for j in range(11, 14))
+    test_idx = tuple((i, j) for i in range(15, 20) for j in range(15, 20))
+
+    data = tuple((i, j) for i in range(1, 20) for j in range(1, 20))
+    index_in_train_idx = np.array([np.where(np.all(np.array(data) == t, axis=1))[0][0] for t in train_idx])
+    index_in_val_idx = np.array([np.where(np.all(np.array(data) == t, axis=1))[0][0] for t in val_idx])
+    index_in_test_idx = np.array([np.where(np.all(np.array(data) == t, axis=1))[0][0] for t in test_idx])
+
+    clp = CustomLabelProcessor(col_name="test_label", label_name="test", id_col="id",
+                               task_type="link_prediction",
+                               train_idx=train_idx, val_idx=val_idx, test_idx=test_idx,
+                               stats_type=None)
+
+    split = clp.data_split(data)
+    assert "train_mask" in split
+    assert "val_mask" in split
+    assert "test_mask" in split
+
+    assert_equal(np.squeeze(np.nonzero(split["train_mask"])), index_in_train_idx)
+    assert_equal(np.squeeze(np.nonzero(split["val_mask"])), index_in_val_idx)
+    assert_equal(np.squeeze(np.nonzero(split["test_mask"])), index_in_test_idx)
+    assert len(split["train_mask"]) == 361
```

Comment:
how 361 is calculated, I am a bit confusing

Reply:
19 * 19, that is the total number of edges.

Reply:
Can you put some comments here?